#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "../protos/E-puck.proto"

WorldInfo {
}
Viewpoint {
  fieldOfView 0.785389
  orientation 0.6003459491766711 0.5316465701668905 -0.5974417676535038 2.1043269451753606
  position -0.12131496794143423 -0.07086748905422416 6.961438270379705
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 0 0 -0.230001
  floorSize 2 2
  floorTileSize 2 2
  floorAppearance PBRAppearance {
    baseColorMap ImageTexture {
      url [
        "map_9-11_4 1.png"
      ]
      filtering 1
    }
    roughness 1
    roughnessMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/chequered_parquetry_roughness.jpg"
      ]
    }
    metalness 0
    IBLStrength 1.00016
    normalMap ImageTexture {
      filtering 1
    }
    normalMapFactor 0
    occlusionMapStrength 2
    emissiveIntensity 0
  }
}
E-puck {
  translation -0.333446 0.49466 -0.229056
  rotation 0 0 1 -3.1415853071795863
  controller "line_mapping"
  turretSlot [
    InertialUnit {
    }
  ]
  groundSensorsSlot [
    DistanceSensor {
      translation 0.026 0 -3.67321e-08
      rotation 0 1 0 1.5708
      name "CEIR"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.01 -0.04 0
      rotation 0 1 0 1.5708
      name "RIRIR"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.01 0.04 0
      rotation 0 1 0 1.5708
      name "LELIR"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 -0.02 0
      rotation 0 1 0 1.5708
      name "RIR"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 0.02 0
      rotation 0 1 0 1.5708
      name "LIR"
      type "infra-red"
    }
  ]
}
